untyped-lambda-calculus/article.md

```diff
@@ -146,63 +146,73 @@ This brings us to the discussion of evaluation strategies.
+The two strategies above are free to go inside abstractions to find redexes.
+For this reason, we call them _strong_.
 
-When we're talking about programming languages, we can't necessarily inspect function bodies, so the weak strategies are considered more practical.
+But there are also strategies that can't go inside abstraction bodies, which we call _weak_.
+When we're talking about programming languages, we can't necessarily inspect function bodies, so the weak strategies are considered more practical there.
 
-It's interesting to note that the call-by-value strategy corresponds to eager evaluation.
+The weak strategy corresponding to the normal order is called _call-by-name_, and the one corresponding to the applicative order is _call-by-value_.
+Call-by-value corresponds to eager evaluation.
 Call-by-name, on the other hand, corresponds to lazy evaluation.
+
 In eager evaluation, arguments are always evaluated before the function call.
 Most programming languages follow this strategy by default.
 In lazy evaluation, some arguments are potentially never evaluated at all, but some computations may be duplicated.
 Very few programming languages implement lazy evaluation this way (the only real example that comes to mind is Algol 60).
 
-A practically significant variation on call-by-name is _call-by-need_, which is, at least in theory, used in Haskell.
+A more practical variation of call-by-name is _call-by-need_, which is, at least in theory, used in Haskell.
 Call-by-need behaves exactly like call-by-name, except that reduction steps that would duplicate computations, don't.
-The trick is to share the computation corresponding to an argument everywhere it appears, such that the shared computation is performed at most once.
+It does that by sharing the computation corresponding to an argument everywhere it appears, so any shared computation is performed at most once.
 
-Expressions that are not reducible using the chosen evaluation strategy are said to be in the _normal form_.
+To sum up, here's a table with all the mentioned strategies and their characteristics:
 
-If a given term has a normal form under both strong strategies, it's the same for both.
-Note that applicative order strategy is strictly weaker than normal order strategy.
-Some terms that normalize under normal order don't under applicative order.
-Likewise for the weak strategies.
+| Name                 | Outermost- or innermost- first? | Strong or weak? |
+|----------------------|---------------------------------|-----------------|
+| Normal order         | Outermost                       | Strong          |
+| Applicative order    | Innermost                       | Strong          |
+| Call-by-name (eager) | Outermost                       | Weak            |
+| Call-by-value (lazy) | Innermost                       | Weak            |
+| Call-by-name         | Innermost                       | Weak            |
```

Comment:
Call-by-name is mentioned twice. I think you meant call-by-need the second time?

A little strange having it in this table here, as it is entirely unclear what differentiates it from call-by-name. It is also considered "lazy evaluation". Actually, under some definition of "lazy", call-by-need is the laziest of them all.

Reply:
The problem is that since it is mentioned, it will also look odd if it doesn't show up. 😅 

Perhaps we can write Call-by-name (Haskell) 🤔?